cardano-api/src/Cardano/Api/ValueParser.hs

```diff
@@ -115,16 +116,17 @@ decimal = do
 assetName :: Parser AssetName
 assetName = do
   hexText <- many hexDigit
-  note "AssetName deserisalisation failed" $
+  failEitherWith
+    (\e -> "AssetName deserisalisation failed: " ++ displayError e) $
     deserialiseFromRawBytesHex AsAssetName $ BSC.pack hexText
 
 -- | Policy ID parser.
 policyId :: Parser PolicyId
 policyId = do
   hexText <- many1 hexDigit
   case textToPolicyId hexText of
-    Just p -> pure p
-    Nothing ->
+    Right p -> pure p
+    Left _ ->
```

Comment:
Why not use `displayError` here as well? Also are policy id's meant to only be 56 hex digits long?